src/map/utils/fishingcontest.cpp

```diff
@@ -0,0 +1,716 @@
+    {
+        return CurrentFishingContest.criteria;
+    }
+
+    uint8 GetContestMeasure()
+    {
+        return CurrentFishingContest.measure;
+    }
+
+    uint16 GetContestFish()
+    {
+        return CurrentFishingContest.fishId;
+    }
+
+    uint32 GetContestStartTime()
+    {
+        return CurrentFishingContest.startTime;
+    }
+
+    uint32 GetContestChangeTime()
+    {
+        return CurrentFishingContest.changeTime;
+    }
+
+    // Contest Control
+    void ProgressContest()
+    {
+        /***********************************************
+         *  Update the phase of the fishing contest
+         *  See enum FISHING_CONTEST_STATUS in fishingutils.h
+         *  CALLED BY ZONE TICK IN SELBINA - DO NOT ADD TO FUNCTION
+         ***********************************************/
+        uint32 currentTime = (uint32)time(NULL);
+        if (currentTime > CurrentFishingContest.changeTime)
+        {
+            FISHING_CONTEST_STATUS newStatus = nextStage[CurrentFishingContest.status];
+
+            if (newStatus == CONTEST_STATUS_RELEASING)
+            {
+                ScoreContest();
+                UpdateRankingHistory();
+            }
+
+            if (newStatus == CONTEST_STATUS_CONTESTING)
+            {
+                InitNewContest();
+            }
+            else
+            {
+                SetContestStatus(newStatus);
+            }
+
+            SetContestChangeTime(GetStageChangeTime((FISHING_CONTEST_STATUS)newStatus, currentTime));
+
+            auto onStageProgress = lua["xi"]["fishingcontest"]["onStageProgress"];
+            if (!onStageProgress.valid())
+            {
+                sol::error err = onStageProgress;
+                ShowError("fishingcontest:onStageProgress: %s", err.what());
+                return;
+            }
+            auto result = onStageProgress(newStatus);
+            if (!result.valid())
+            {
+                sol::error err = result;
+                ShowError("fishingcontest:ProgressContest:(onStageProgress): %s", err.what());
+                return;
+            }
+        }
+    }
+
+    void UpdateRankingHistory()
+    {
+        // This should only be run once per contest, once the contest progresses from "Accepting" to "Releasing"
+        // Traverse the entries and update the database for each player
+        for (FishingContestEntry* it : FishingContestEntries)
+        {
+            // Get the Player ID
+            uint32 charID = 0;
+            auto   ret    = _sql->Query("SELECT charid FROM chars WHERE charname = '%s' LIMIT 1", it->name);
+            if (ret != SQL_ERROR && _sql->NumRows() != 0 && _sql->NextRow() == SQL_SUCCESS)
+            {
+                charID = _sql->GetUIntData(0);
+            }
+
+            if (charID > 0)
+            {
+                // Get the rank for column header
+                uint8 rankGroup = 0;
+                if (it->contestRank > 0 and it->contestRank <= 4)
```

Comment:
we don't use verbal `and` or `or` (I don't know how CI hasn't caught all of this...), use the ampersand and pipe versions